### DISCUSSION TITLE
[Question]: How to tell if Live View is using WebRTC or MSE?

### DISCUSSION THREAD

**Discussion by Orochikaku** [blakeblackshear/frigate#13243] [Ask A Question] [answered]
### What is your question:

I'm [setting up a reverse proxy](https://github.com/blakeblackshear/frigate/discussions/13131#discussion-7062018) and along the way I'm setting up ufw as well.

In the processes of doing so I've allowed incoming connections on port 22 for SSH, 443 for https and 8554 for the RTSP re-stream feeds. 

My question now is how do I ensure that WebRTC is being used still? I opened port 1984 on docker and checked on the go2rtc UI that WebRTC seems to be working but I just want some indication that it's working in the Frigate Live UI as well.

---

- [docker-compose.yml](https://gist.github.com/Orochikaku/3d9eea5147099462aff143ba547a0bac)
- [config.yaml](https://gist.github.com/Orochikaku/f7c66a6bac425bc193bb68ddf0b0bc6a)
- [Hardware](https://lucid.app/lucidchart/f6c7fc39-e502-468d-b965-fda250901d86/view)

Note: I have a non-static IP

---

Thank you in advance :)

**Comment by hawkeye217** [comment, marked as answer]
Frigate intelligently selects the live view type based on a number of factors. There’s no simple indication in the UI as to what it’s using at a moment in time. But if you really wanted to discover it, you could view the nginx logs to see the path pulled by the live player. It will include the stream type in the URL.

**Reply by Orochikaku** [reply]
Thanks that works for me! However, while reviewing the logs I found that Frigate had only selected MSE. Is there any way I could figure out if it's simply because Frigate decided MSE was the best available option or whether it was because it failed to connect via WebRTC?

**Reply by hawkeye217** [reply]
MSE is what Frigate defaults to as it's the most compatible with modern browsers and doesn't require additional configuration like WebRTC does. If MSE fails for some reason, WebRTC is attempted. If that fails, the player falls back to jsmpeg. 

The latest dev builds also have optimizations for MSE to improve latency in long-running live views and slow network conditions, so there really is little to no difference between the MSE and WebRTC streams.

**Reply by Orochikaku** [reply]
Ah I see, thank you for the clarification. I'm sure lot's of people have been asking about this. In my case it was due to the [documentation ](https://github.com/blakeblackshear/frigate/blob/22fe261dd6ea9d90d803c88214e408fd429a288d/docs/docs/configuration/live.md)making it seem as though WebRTC is preferable over MSE.

**Reply by hawkeye217** [reply]
Thanks. I'll make a docs update. With the latest changes to MSE, there is virtually no difference.
